contracts/Pool.sol

```diff
@@ -0,0 +1,381 @@
+  **/
+  function setRatesCalculator(IRatesCalculator ratesCalculator_) external onlyOwner {
+    require(address(ratesCalculator_) != address(0), "The rates calculator cannot set to a null address");
+    _ratesCalculator = ratesCalculator_;
+    _updateRates();
+  }
+
+
+  /**
+   * Sets the new borrowers registry contract.
+   * The borrowers registry decides if an account can borrow funds.
+   * Only the owner of the Contract can execute this function.
+   * @dev _borrowersRegistry the address of borrowers registry
+  **/
+  function setBorrowersRegistry(IBorrowersRegistry borrowersRegistry_) external onlyOwner {
+    require(address(borrowersRegistry_) != address(0), "The borrowers registry cannot set to a null address");
+
+    _borrowersRegistry = borrowersRegistry_;
+  }
+
+
+  /* ========== MUTATIVE FUNCTIONS ========== */
+  function transfer(address recipient, uint256 amount) external override returns (bool) {
+    require(recipient != address(0), "ERC20: cannot transfer to the zero address");
+    require(recipient != address(this), "ERC20: cannot transfer to the pool address");
+
+    _accumulateDepositInterest(msg.sender);
+
+    require(_deposited[msg.sender] >= amount, "ERC20: transfer amount exceeds balance");
+
+    // (this is verified in "require" above)
+    unchecked {
+      _deposited[msg.sender] -= amount;
+    }
+
+    _accumulateDepositInterest(recipient);
+    _deposited[recipient] += amount;
+
+    emit Transfer(msg.sender, recipient, amount);
+
+    return true;
+  }
+
+  function allowance(address owner, address spender) external view override returns (uint256) {
+    return _allowed[owner][spender];
+  }
+
+  function approve(address spender, uint256 amount) external override returns (bool) {
+    _accumulateDepositInterest(msg.sender);
+
+    require(_deposited[msg.sender] >= amount, "ERC20: approve amount exceeds balance");
+    _allowed[msg.sender][spender] = amount;
+
+    emit Approval(msg.sender, spender, amount);
+
+    return true;
+  }
+
+  function transferFrom(address sender, address recipient, uint256 amount) external override returns (bool) {
+    require(amount <= _allowed[sender][msg.sender], "Not enough tokens allowed to transfer required amount.");
+    require(recipient != address(0), "ERC20: cannot transfer to the zero address");
+    require(recipient != address(this), "ERC20: cannot transfer to the pool address");
+
+    _accumulateDepositInterest(msg.sender);
+
+    require(amount <= _deposited[sender], "Not enough tokens to transfer required amount.");
+
+    _deposited[sender] -= amount;
+    _allowed[sender][msg.sender] -= amount;
+
+    _accumulateDepositInterest(recipient);
+    _deposited[recipient] += amount;
+
+    emit Transfer(sender, recipient, amount);
+
+    return true;
+  }
+
+
+  /**
+   * Deposits the message value
+   * It updates user deposited balance, total deposited and rates
+  **/
+  function deposit() payable virtual external {
```

Comment:
I don't see any potential vulnerability but could we add nonReentrant just in case @piotruda ?

Reply:
done